lib/safe_code/validator.ex

```diff
@@ -25,6 +25,18 @@ defmodule SafeCode.Validator do
     |> validate_quoted(opts)
   end
 
+  @spec validate_heex(binary, keyword) :: {:ok, ast :: Macro.t()} | {:error, Exception.t()}
+  def validate_heex(heex, opts \\ []) when is_binary(heex) do
+    quoted =
+      heex
+      |> HeexParser.parse_template()
+      |> validate_quoted(include_phoenix(opts))
```

Comment:
Lets call directly into the other one, that way these two won't drift if a change is made to the parsing logic in one and doesn't get copied to the other version.
```suggestion
    quoted = validate_heex!(heex, opts)
```